sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -472,12 +496,25 @@ public async virtual Task CleanupAbortedJobPartAsync()
             // If the failure occurred due to the file already existing or authentication,
             // and overwrite wasn't enabled, don't delete the existing file. We can remove
             // the unfinished file for other error types.
-            if (JobPartFailureType.Other == _failureType)
+            // If a Pause was called, we can remove the unfinished file.
+            if (JobPartFailureType.Other == _failureType || DataTransferState.Pausing == JobPartStatus.State)
             {
                 // If the job part is paused or ended with failures
                 // delete the destination resource because it could be unfinished or corrupted
                 // If we resume we would have to start from the beginning anyways.
-                await _destinationResource.DeleteIfExistsAsync(_cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    // We can't pass the cancellation token
+                    // here due to the fact that the job's cancellation token has already been called.
+                    // We are cleaning up / deleting optimistically, which means that if
+                    // the clean / delete attempt fails, then we continue on. The failure may be due
+                    // to the overall reason why the job was cancelled in the first place.
+                    await _destinationResource.DeleteIfExistsAsync().ConfigureAwait(false);
+                }
+                finally
+                {
+                    // We are cleaning up / deleting optimistically, move on if it fails.
+                }
```

Comment:
This try/finally does not do anything. I think you would want `catch (Exception)`. As is, an exception in delete would be thrown from here.

Reply:
That wouldn't be deleting optimistically. Do we actually want to log / invoke arg failures here? In the state that a transfer has no failures and then a Pause occurs, then this is the only failure, should we be invoking a failure here? Also this would change the DataTransferStatus.HasFailures to `true`. Also during clean up should there be logged failures?